src/json/json.cc

```diff
@@ -624,6 +731,56 @@ int Command_JsonSet(ValkeyModuleCtx *ctx, ValkeyModuleString **argv, int argc) {
     return ValkeyModule_ReplyWithSimpleString(ctx, "OK");
 }
 
+int Command_JsonMSet(ValkeyModuleCtx *ctx, ValkeyModuleString **argv, int argc) {
+    ValkeyModule_AutoMemory(ctx);
+
+    MSetCmdArgsList args;
+    JsonUtilCode rc = parseAndValidateMSetCmdArgs(ctx, argv, argc, &args);
+    if (rc != JSONUTIL_SUCCESS) {
+        if (rc == JSONUTIL_WRONG_NUM_ARGS)
+            return ValkeyModule_WrongArity(ctx);
+        else
+            return ValkeyModule_ReplyWithError(ctx, jsonutil_code_to_message(rc));
+    }
+
+    // Apply changes
+    for (size_t i = 0; i < args.num_keys; i++) {
+
+        if (args.args_list[i].is_root_path){  // root doc
+            // parse incoming JSON string
+            JDocument *doc;
+            rc = dom_parse(ctx, args.args_list[i].json, args.args_list[i].json_len, &doc);
+            ValkeyModule_Assert(rc == JSONUTIL_SUCCESS);
+
+            dom_set_doc_size(doc, args.args_list[i].doc_size);
+
+            // set Valkey key
+            ValkeyModule_ModuleTypeSetValue(args.args_list[i].key, DocumentType, doc);
+
+            // update stats
+            jsonstats_update_stats_on_insert(doc, true, 0, args.args_list[i].doc_size, args.args_list[i].doc_size);
+        } else {
+            JDocument *doc = static_cast<JDocument*>(ValkeyModule_ModuleTypeGetValue(args.args_list[i].key));
+            size_t orig_doc_size = dom_get_doc_size(doc);
+            
+            rc = dom_set_value(ctx, doc, args.args_list[i].path, args.args_list[i].json, 
+                             args.args_list[i].json_len, false, false);
+            ValkeyModule_Assert(rc == JSONUTIL_SUCCESS);
+
+            dom_set_doc_size(doc, args.args_list[i].doc_size);
```

Comment:
Likewise, we must track the memory usage here instead of inside parseAndValidateMSetCmdArgs.